TACC/protx-dashboard
# Task/cooks 328 new analytics report top 3 features per county

## Overview: ##
When a county is selected in Analytics, show the counties top 3 maltreatment risk factors with their associated rank (1-3), demographic feature name, and whether the feature has a negative or positive correlation to increased maltreatment

## Related Jira tickets: ##

* [COOKS-328](https://jira.tacc.utexas.edu/browse/COOKS-328)

## Summary of Changes: ##

Created new Analytics Predictive Table component. Added to the chart instructions component and made a new chart type to have the displayed explanation of the table provided by the data team. Updated the api.py file to catch an error if no data is available for that selected county.

## Testing Steps: ##
1. Go to https://cep.test/protx/dash/analytics and choose a county
2. Click around until you find a county with no data to ensure the proper display is shown. Make sure the data and table populates for counties that do have top three risk demographic data as well.


## UI Photos:
Display when there's no data for that county
<img width="1785" alt="Screen Shot 2022-10-20 at 12 54 30 PM" src="https://user-images.githubusercontent.com/96220951/197022772-7559e521-7b9e-4412-81f8-abf2fd82177f.png">



Display when there is data for that county

<img width="1777" alt="Screen Shot 2022-10-20 at 12 53 55 PM" src="https://user-images.githubusercontent.com/96220951/197022828-1338b941-40d2-4228-aece-5eae77c258d5.png">


## Notes: ##
For @nathanfranklin :

Should I change the protx-client/src/redux/reducers/protx.reducers.js file back to 
`export const initialAnalyticsState = {
  loading: true,`

Wasn't sure if this was a bug or if it was just for our testing yesterday. Thank you!
